aws-datastore/src/test/java/com/amplifyframework/datastore/appsync/AppSyncExtensionsTest.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.datastore.appsync;
+
+import com.google.gson.internal.LinkedTreeMap;
```

Comment:
The `AppSyncExtensions` utility and its test are "above" the abstraction of Gson, at this point. So, we should not import any Gson components as that's a leaking abstraction. If you need a map which preserves insertion order, consider a `LinkedHashMap`, or if you need one that preserves a natural ordering of keys, look at a (Java Collections) `TreeMap`.

Reply:
Good point!  No need to preserve order here, so I'll just use a `HashMap`.  

I also just realized that while `GsonErrorDeserializer` is serializing `extensions` as a `HashMap`, all `JsonObject`s _inside_ of `extensions` are serialized as Gson `LinkedTreeMap`s, since that is the default when using gson.  I will update this to make sure `HashMap`s are used.

Reply:
Oh! Good catch. Getting into the fine points here :-D